src/index.js

```diff
@@ -1,31 +1,38 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
+   //... your code goes here
   console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  const price = product.querySelector('.price span').innerText;
+  const quantity = product.querySelector('.quantity input').value;
+  const subtotal = product.querySelector('.subtotal span');
+  const subtotalText = price * quantity;
+  subtotal.innerText = subtotalText;
+  return subtotalText
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
-
-  // ITERATION 3
-  //... your code goes here
+  
+  const products = document.getElementsByClassName('product');
+    let total = 0;
+    for (let i=0;i<products.length;i++){
+      total += updateSubtotal(products[i]);
+    }
+    document.querySelector('#total-value span').innerText = total;
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
-  const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  const target1 = event.currentTarget;
+  console.log('The target in remove is:', target1);
+
+  
+const target2 = event.currentTarget;
```

Comment:
Another way to do this is using the parentNode property, like:

`target.parentNode.parentNode.remove();`

This way, you're targeting the `tr` with `class = product` and then using the remove method to remove it!

Another thing you could've added is the calculateAll function to calculate all the prices after you remove one product.